src/components/BalancePage/BalancePage.jsx

```diff
@@ -4,22 +4,42 @@ import { HOME } from '../../routes';
 
 import NavLink from '../NavLinks/NavLink.jsx';
 
-const BalancePage = () => (
-  <div className="balance__page">
-    <h1 className="balance__page_balance">
-      Your current balance is:
-      <span className="main__page_header-span"> 0,00 💲</span>
-    </h1>
-    <h3 className="balance__page_spent">
-      Total spent:
-      <span className="main__page_header-span"> 1034074028,00 💲</span>
-    </h3>
-    <h3 className="balance__page_received">
-      Total received:
-      <span className="main__page_header-span"> 1034074028,00 💲</span>
-    </h3>
-    <NavLink text={`Home`} route={HOME} />
-  </div>
-);
+const { useSelector } = require('react-redux');
+
+const BalancePage = () => {
+  const list = useSelector(state => state.transactionsList.transactionsList);
+
+  const totalInputs = list
+    .map(trns => trns.inputs)
+    .flat()
+    .map(input => input.output_value)
+    .reduce((inVal, inAcc) => inVal + inAcc);
+
+  const totalOutputs = list
+    .map(trns => trns.outputs)
+    .flat()
+    .map(output => output.value)
+    .reduce((outVal, outAcc) => outVal + outAcc);
+
```

Comment:
looks like calculating `totalInputs ` and `totalOutputs ` pretty much the same, could you move it to the helper function, please?

Reply:
Done. Can you please check?